File: experiments/src/programs/early/Early.java
public class Early {

  public static void commitEarly(int a, int b) {
    int c = a * b;
    if (b > 2 && b == c) {
      System.out.println("Solved early commitment");
    }
  }

  public static void main(String[] args) {
    int a = 1, b = 3;
    commitEarly(a, b);
  }
}

File: experiments/src/programs.c#/EarlyTest/Early.cs
/*
 * Automatically converted from Java to C# using Sharpen.
 */

using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Pex.Framework;

[TestClass]
public partial class Early
{
    [PexMethod(MaxConstraintSolverTime = 5, MaxRunsWithoutNewTests = 500)]
    public static void CommitEarly(int a, int b)
	{
		int c = a * b;
		if (b > 2 && b == c)
		{
			System.Console.Out.WriteLine("Solved early commitment");
		}
	}
}
